library/src/main/scala/scala/scalajs/runtime/package.scala

```diff
@@ -97,7 +98,11 @@ package object runtime {
    *
    *  See [[LinkingInfo]] for details.
    */
-  def linkingInfo: LinkingInfo = throw new Error("stub")
+  @nowarn @deprecated("Use scala.scalajs.LinkingInfo or scala.scalajs.js.special.fileLevelThis instead", "1,17,0")
+  def linkingInfo: LinkingInfo =
+    throw new UnsupportedOperationException("scala.scalajs.runtime.linkingInfo is deprecated," +
+      "use scala.scalajs.LinkingInfo instead. " +
+      "For fileLevelThis, use scala.scalajs.js.special.fileLevelThis.")
```

Comment:
If the method is only `@deprecated` instead of being entirely removed, we must still provide a correct implementation. It could be a `new LinkingInfo { ... }` with `val`s initialized from the corresponding `LinkingInfo` fields.